xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererDX.cpp

```diff
@@ -229,30 +229,16 @@ COverlayImageDX::COverlayImageDX(CDVDOverlayImage* o)
 
   if(o->source_width && o->source_height)
   {
-    float center_x = (float)(0.5f * o->width  + o->x) / o->source_width;
-    float center_y = (float)(0.5f * o->height + o->y) / o->source_height;
+    float scareheight = (float)sourceHeight * o->source_width / sourceWidth;
+    float center_x = (0.5f * o->width  + o->x) / o->source_width;
```

Comment:
I don't follow this math here. please elaborate. You don't scale m_width, which seems not to be correct.

Reply:
the method is, first, zoom videosourcesize to the same width to the subsourcesize. then calculate y position in videosourcesize coordinate system.

Reply:
in order to keep AR we need to scale x and y with the same factor

Reply:
the goal here is not stretch subsourcesize to videosourcesize, is change coordinate system from subview to videoview.

Reply:
think about change coordinate system from 1920x1080 to 1920x800, only y position need change.

Reply:
both m_width and m_height not changed. only veiw size height changed here.

Reply:
example: subSource 1920x1080, videoSource 720x576
this code will badly fail in this case
you have to scale to video size here because we set ALIGN_VIDEO

Reply:
Maybe I'm missing something here, but why should you get 16:9 bitmap subs on a 4:3 video? Or is that "just in case"?

Reply:
kodi has a multi source demuxer and subs may come from a different source. but even if the smaller size had same AR the current code would fail because it does not consider scaling of all dimensions.

Reply:
> kodi has a multi source demuxer and subs may come from a different source

I know, but realistically - where are you going to get 4:3 bitmap subs for a 16:9 video from? Well anyway, disregard; I'm probably thinking too narrowly...

Would be nice to have samples/test files for all the strange cases like this though so we can verify whether everything works correctly.

Reply:
You can add any external subtitle file to any video by giving the subtitle file the same filename as the video file. The subs won't make any sense for this video but good for testing rendering.

Reply:
know the problem now, thanks

Reply:
both subsource and videosource may not be 1080P size. then need to simulate sub and video play on like 1920x1080 screen, the trans the coordinate system.

Reply:
I don't think there's any reason to have 1920x1080 hardcoded anywhere. Use the information you have for calculations.

Reply:
@pkerling the sub file which not align in videosourcesize usually make to align in 1.78:1 screen. so simulate a 1920x1080 screen can correct positioning sub with video. may be use 1.78x1 here?